From 2a52234a103c8a7bf280e999484c3860d44f9b3a Mon Sep 17 00:00:00 2001
From: Ido Reis <idor@ti.com>
Date: Thu, 21 Jul 2011 22:20:03 +0300
Subject: [PATCH] add if and dhcp services support to hostapd start/stop methods

usage of setprop for invokation of two services to perform ifconfig and dhcp deamon when starting ap.
these services are stopped when stoping the ap.

changes to init.omap4430.rc are necessary in order to perform make the services to actually do somthing.
we added these services since the hostapd_bin did not configured the ip and no dhcp deamon was ran after srating it.

Change-Id: I2d236894416f972b2664b456f30434cddf534223
---
 SoftapController.cpp |   18 ++++++++++++++++++
 SoftapController.h   |    2 ++
 2 files changed, 20 insertions(+), 0 deletions(-)

diff --git a/SoftapController.cpp b/SoftapController.cpp
index 1de311e..ebe470a 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -324,10 +324,28 @@ int SoftapController::startHostapd() {
     LOGD("hostapd started OK");
     mHostapdStarted = true;
 
+    if (property_set("ctl.start", HOSTAPD_IFCFG_SERVICE_NAME) < 0) {
+        LOGE("Failed to start ifcfg");
+        return -1;
+    }
+
+    if (property_set("ctl.start", HOSTAPD_DHCPD_SERVICE_NAME) < 0) {
+            LOGE("Failed to start dhcpd");
+            return -1;
+    }
+
     return 0;
 }
 
 int SoftapController::stopHostapd() {
+	if (property_set("ctl.stop", HOSTAPD_DHCPD_SERVICE_NAME) < 0) {
+		LOGE("Failed to stop dhcpd");
+		return -1;
+	}
+	if (property_set("ctl.stop", HOSTAPD_IFCFG_SERVICE_NAME) < 0) {
+		LOGE("Failed to stop ifcfg");
+		return -1;
+	}
 
     if (property_set("ctl.stop", HOSTAPD_SERVICE_NAME) < 0) {
         LOGE("Failed to stop hostapd service");
diff --git a/SoftapController.h b/SoftapController.h
index f56ac59..4d4c2d0 100644
--- a/SoftapController.h
+++ b/SoftapController.h
@@ -32,6 +32,8 @@
 #include "nl80211.h"
 
 #define HOSTAPD_SERVICE_NAME "hostapd_bin"
+#define HOSTAPD_IFCFG_SERVICE_NAME "ifcfg_softap"
+#define HOSTAPD_DHCPD_SERVICE_NAME "dhcpd_softap"
 #define HOSTAPD_STATE_PROP "init.svc." HOSTAPD_SERVICE_NAME
 #define AP_WAKE_LOCK "hotspot_wake_lock"
 
-- 
1.7.0.4

